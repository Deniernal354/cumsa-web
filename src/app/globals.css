@import "tailwindcss";

:root {
  --background: #f8fafc;
  --foreground: #0f172a;
  --surface: #ffffff;
  --muted: #64748b;
  --navy: #2a7487;
  --blue: #5aced8;
  --red: #c73221;
  --header-top: #123a66;
}

body {
  background: var(--background);
  color: var(--foreground);
  font-family: Arial, Helvetica, sans-serif;
}

a {
  text-underline-offset: 4px;
}

.link-cta {
  color: var(--blue);
  font-weight: 700;
  text-decoration: underline;
  text-underline-offset: 3px;
}

.link-cta:hover {
  color: var(--red);
}

.btn-cozy {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  border-radius: 0.6rem;
  font-weight: 600;
  font-size: 0.9rem;
  transition:
    transform 150ms ease,
    box-shadow 150ms ease,
    opacity 150ms ease;
}

.btn-cozy:hover {
  transform: translateY(-1px);
  box-shadow: 0 6px 16px rgba(0, 0, 0, 0.08);
}

.btn-primary {
  background: var(--blue);
  color: white;
}

.btn-outline {
  background: var(--surface);
  color: var(--foreground);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

:focus-visible {
  outline: 2px solid color-mix(in oklab, var(--blue) 70%, white);
  outline-offset: 2px;
}

.container-base {
  margin-left: auto;
  margin-right: auto;
  max-width: 80rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

@media (min-width: 640px) {
  .container-base {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media (min-width: 1024px) {
  .container-base {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

.transition-base {
  transition: all 200ms ease-out;
}

.card {
  background: var(--surface);
  border: 1px solid rgba(0, 0, 0, 0.12);
  border-radius: 0.75rem;
}

.title {
  color: var(--red);
}

.tint-blue {
  filter: invert(28%) sepia(88%) saturate(3612%) hue-rotate(211deg) brightness(93%) contrast(98%);
}

.logo-mask-blue {
  background-color: var(--blue);
  -webkit-mask-image: url("/CU-MSA-LOGO-blank-2.png");
  mask-image: url("/CU-MSA-LOGO-blank-2.png");
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
  -webkit-mask-position: center;
  mask-position: center;
  -webkit-mask-size: contain;
  mask-size: contain;
  display: inline-block;
}
