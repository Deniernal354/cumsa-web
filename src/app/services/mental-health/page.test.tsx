import { test, expect } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import MentalHealthPage from "./page";

test("renders key mental health sections", () => {
  render(<MentalHealthPage />);
  const mainHeading = screen.getByRole("heading", { name: /Mental Health & Wellâ€‘Being/i });
  const emergency = screen.getByRole("heading", { name: /Immediate Help/i });
  const hotlines = screen.getByRole("heading", { name: /Hotlines/i });
  const revert = screen.getByRole("heading", { name: /Revert Support/i });
  expect(mainHeading.textContent).toMatch(/Mental Health/i);
  expect(emergency).not.toBeNull();
  expect(hotlines).not.toBeNull();
  expect(revert).not.toBeNull();
});

test("matches snapshot", () => {
  const { container } = render(<MentalHealthPage />);
  expect(container).toMatchSnapshot();
});
