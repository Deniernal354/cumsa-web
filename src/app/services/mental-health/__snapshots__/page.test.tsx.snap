// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <main
    class="mx-auto max-w-5xl px-5 py-10 flex flex-col gap-10"
  >
    <header
      class="text-center flex flex-col gap-3"
    >
      <h1
        class="text-4xl font-semibold tracking-tight"
      >
        Mental Health & Well‑Being
      </h1>
      <p
        class="text-neutral-600 text-sm max-w-2xl mx-auto"
      >
        Gentle, faith-informed pointers. This page curates quick help, calm steps, and trusted places to learn & reach out.
      </p>
    </header>
    <section
      aria-labelledby="mh-emergency"
      class="grid md:grid-cols-2 gap-6"
    >
      <div
        class="rounded-xl border border-neutral-200/60 bg-white/70 backdrop-blur shadow-sm hover:shadow-md transition-shadow p-5 flex flex-col gap-3 bg-rose-50/70 border-rose-200"
      >
        <h2
          class="text-xl font-bold flex items-center gap-2 text-rose-700"
          id="mh-emergency"
        >
          <span
            aria-hidden="true"
          >
            ❗
          </span>
           Immediate Help
        </h2>
        <p
          class="text-sm text-rose-800 leading-snug"
        >
          If you or someone is in danger or at risk of harming themselves: call local emergency services right now. On campus, contact security or the counselling centre urgent line.
        </p>
        <ul
          class="text-sm flex flex-col gap-1 mt-1"
        >
          <li>
            Emergency: 
            <span
              class="font-medium"
            >
              911
            </span>
          </li>
          <li>
            Campus Security: 613‑520‑4444
          </li>
        </ul>
        <p
          class="text-[11px] text-rose-600 mt-auto"
        >
          You are not alone. Help now is better than silence.
        </p>
      </div>
      <div
        class="rounded-xl border border-neutral-200/60 bg-white/70 backdrop-blur shadow-sm hover:shadow-md transition-shadow p-5 flex flex-col gap-3"
      >
        <h2
          class="text-xl font-bold flex items-center gap-2"
        >
          <span
            aria-hidden="true"
          >
            🧭
          </span>
           Quick Calm
        </h2>
        <ul
          class="text-sm leading-relaxed grid grid-cols-2 gap-x-4 gap-y-1"
        >
          <li>
            3 deep belly breaths
          </li>
          <li>
            Cold water on wrists
          </li>
          <li>
            Stand + stretch spine
          </li>
          <li>
            Look for 5 colors
          </li>
          <li>
            Make wudu mindfully
          </li>
          <li>
            Recite Ayat al-Kursi
          </li>
          <li>
            Message a friend
          </li>
          <li>
            Step outside light
          </li>
        </ul>
        <p
          class="text-xs text-neutral-500"
        >
          Tiny resets stack into relief.
        </p>
      </div>
    </section>
    <section
      aria-labelledby="mh-hotlines"
      class="grid lg:grid-cols-3 md:grid-cols-2 gap-6"
    >
      <div
        class="rounded-xl border border-neutral-200/60 bg-white/70 backdrop-blur shadow-sm hover:shadow-md transition-shadow p-5 flex flex-col gap-3"
      >
        <h2
          class="text-xl font-bold flex items-center gap-2"
          id="mh-hotlines"
        >
          <span
            aria-hidden="true"
          >
            📞
          </span>
           Hotlines
        </h2>
        <ul
          class="text-sm flex flex-col gap-1"
        >
          <li>
            Nisa (women) Helpline: 
            <span
              class="font-medium"
            >
              1‑888‑456‑8043
            </span>
          </li>
          <li>
            Naseeha Youth: 
            <span
              class="font-medium"
            >
              1‑866‑627‑3342
            </span>
          </li>
          <li>
            Talk Suicide Canada: 9-8-8 
          </li>
          <li>
            National Child Abuse: 1‑800‑422‑4453
          </li>
        </ul>
        <p
          class="text-[11px] text-neutral-500 mt-1"
        >
          Availability may change; check before calling.
        </p>
      </div>
      <div
        class="rounded-xl border border-neutral-200/60 bg-white/70 backdrop-blur shadow-sm hover:shadow-md transition-shadow p-5 flex flex-col gap-3"
      >
        <h2
          class="text-xl font-bold flex items-center gap-2"
        >
          <span
            aria-hidden="true"
          >
            📚
          </span>
           Learn & Reflect
        </h2>
        <ul
          class="text-sm flex flex-col gap-1"
        >
          <li>
            <a
              class="text-blue-600 hover:text-blue-700 underline decoration-dotted underline-offset-2"
              href="https://yaqeeninstitute.ca/read/psychology-mental-health"
              rel="noopener noreferrer"
              target="_blank"
            >
              Yaqeen: Psychology & Mental Health
            </a>
          </li>
          <li>
            <a
              class="text-blue-600 hover:text-blue-700 underline decoration-dotted underline-offset-2"
              href="https://seekersguidance.org/answers"
              rel="noopener noreferrer"
              target="_blank"
            >
              SeekersGuidance: Answers (search mental health)
            </a>
          </li>
          <li>
            <a
              class="text-blue-600 hover:text-blue-700 underline decoration-dotted underline-offset-2"
              href="https://muslimmentalhealth.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Muslim Mental Health (directory)
            </a>
          </li>
        </ul>
        <p
          class="text-[11px] text-neutral-500 mt-auto"
        >
          Knowledge can soften stigma.
        </p>
      </div>
      <div
        class="rounded-xl border border-neutral-200/60 bg-white/70 backdrop-blur shadow-sm hover:shadow-md transition-shadow p-5 flex flex-col gap-3"
      >
        <h2
          class="text-xl font-bold flex items-center gap-2"
        >
          <span
            aria-hidden="true"
          >
            🧠
          </span>
           Professional & Community
        </h2>
        <ul
          class="text-sm flex flex-col gap-1"
        >
          <li>
            <a
              class="text-blue-600 hover:text-blue-700 underline decoration-dotted underline-offset-2"
              href="https://carleton.ca/health/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Campus Health & Counselling (Carleton)
            </a>
          </li>
          <li>
            <a
              class="text-blue-600 hover:text-blue-700 underline decoration-dotted underline-offset-2"
              href="https://mfso.ca/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Muslim Family Services (MFSO)
            </a>
          </li>
        </ul>
        <p
          class="text-[11px] text-neutral-500 mt-auto"
        >
          Getting help is strength, not failure.
        </p>
      </div>
    </section>
    <section
      aria-labelledby="mh-reverts"
      class="grid md:grid-cols-2 gap-6"
    >
      <div
        class="rounded-xl border border-neutral-200/60 bg-white/70 backdrop-blur shadow-sm hover:shadow-md transition-shadow p-5 flex flex-col gap-3"
      >
        <h2
          class="text-xl font-bold flex items-center gap-2"
          id="mh-reverts"
        >
          <span
            aria-hidden="true"
          >
            🌱
          </span>
           Revert Support
        </h2>
        <p
          class="text-sm leading-snug"
        >
          New to Islam? Identity shifts can feel tender. You're allowed time, questions, and boundaries. Small, consistent acts over intensity.
        </p>
        <ul
          class="text-sm flex flex-col gap-1"
        >
          <li>
            Pair with a gentle mentor (reach out to us)
          </li>
          <li>
            <a
              class="text-blue-600 hover:text-blue-700 underline decoration-dotted underline-offset-2"
              href="https://yaqeeninstitute.ca/read/psychology-mental-health"
              rel="noopener noreferrer"
              target="_blank"
            >
              Foundational articles (Mental Health)
            </a>
          </li>
          <li>
            <a
              class="text-blue-600 hover:text-blue-700 underline decoration-dotted underline-offset-2"
              href="https://newmuslimacademy.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              New Muslim Academy
            </a>
          </li>
          <li>
            <a
              class="text-blue-600 hover:text-blue-700 underline decoration-dotted underline-offset-2"
              href="https://seekersguidance.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              SeekersGuidance classes
            </a>
          </li>
        </ul>
        <p
          class="text-[11px] text-neutral-500"
        >
          Faith grows in kindness to self.
        </p>
      </div>
      <div
        class="rounded-xl border border-neutral-200/60 bg-white/70 backdrop-blur shadow-sm hover:shadow-md transition-shadow p-5 flex flex-col gap-3"
      >
        <h2
          class="text-xl font-bold flex items-center gap-2"
        >
          <span
            aria-hidden="true"
          >
            🪴
          </span>
           Gentle Reminders
        </h2>
        <ul
          class="text-sm flex flex-col gap-1"
        >
          <li>
            Du'a is heard, even if your words are few
          </li>
          <li>
            Rest, sunlight, and gentle meals help more than you think
          </li>
          <li>
            Your feelings are real, but not every thought is true
          </li>
          <li>
            Set small goals and notice your progress
          </li>
          <li>
            Faith and therapy can work together
          </li>
        </ul>
        <p
          class="text-[11px] text-neutral-500 mt-auto"
        >
          Mercy over perfection.
        </p>
      </div>
    </section>
    <section
      aria-labelledby="mh-disclaimer"
      class="text-center flex flex-col gap-2"
    >
      <p
        class="text-xs text-neutral-500 max-w-2xl mx-auto leading-relaxed"
        id="mh-disclaimer"
      >
        This page is an evolving draft, not medical or clinical advice. Consult qualified professionals for diagnosis or treatment. Suggest edits or additions by contacting us.
      </p>
      <a
        class="text-xs text-blue-600 hover:text-blue-700 underline decoration-dotted"
        href="/contact-us"
      >
        Share a resource / feedback
      </a>
    </section>
  </main>
</div>
`;
